nexus/tests/integration_tests/crucible_replacements.rs

```diff
@@ -0,0 +1,577 @@
+        .expect("failed to delete disk");
+
+    // Make sure that all the background tasks can run to completion.
+
+    run_replacement_tasks_to_completion(&internal_client).await;
+
+    // Assert the request is in state Complete
+
+    let region_replacement = datastore
+        .get_region_replacement_request_by_id(&opctx, replacement_request_id)
+        .await
+        .unwrap();
+
+    assert_eq!(
+        region_replacement.replacement_state,
+        RegionReplacementState::Complete,
+    );
+
+    // Assert there are no more Crucible resources
+
+    assert!(disk_test.crucible_resources_deleted().await);
+}
+
+/// Assert that a region replacement request in state "Running" can have its
+/// volume deleted and still transition to Complete
+#[nexus_test]
+async fn test_delete_volume_region_replacement_state_running(
+    cptestctx: &ControlPlaneTestContext,
+) {
+    let nexus = &cptestctx.server.server_context().nexus;
+    let datastore = nexus.datastore();
+    let opctx =
+        OpContext::for_tests(cptestctx.logctx.log.new(o!()), datastore.clone());
+    let client = &cptestctx.external_client;
+    let internal_client = &cptestctx.internal_client;
+
+    // Create four zpools, each with one dataset. This is required for region
+    // and region snapshot replacement to have somewhere to move the data.
+    let sled_id = cptestctx.first_sled();
+    let disk_test = DiskTestBuilder::new(&cptestctx)
+        .on_specific_sled(sled_id)
+        .with_zpool_count(4)
+        .build()
+        .await;
+
+    // Create a disk
+    let _project_id = create_project_and_pool(client).await;
+
+    let disk = create_disk(&client, PROJECT_NAME, "disk").await;
+
+    // Manually create the region replacement request
+
+    let (.., db_disk) = LookupPath::new(&opctx, &datastore)
+        .disk_id(disk.identity.id)
+        .fetch()
+        .await
+        .unwrap();
+
+    assert_eq!(db_disk.id(), disk.identity.id);
+
+    let disk_allocated_regions =
+        datastore.get_allocated_regions(db_disk.volume_id).await.unwrap();
+    let (_, region) = &disk_allocated_regions[0];
+
+    let replacement_request_id = datastore
+        .create_region_replacement_request_for_region(&opctx, &region)
+        .await
+        .unwrap();
+
+    // Assert the request is in state Requested
+
+    let region_replacement = datastore
+        .get_region_replacement_request_by_id(&opctx, replacement_request_id)
+        .await
+        .unwrap();
+
+    assert_eq!(
+        region_replacement.replacement_state,
+        RegionReplacementState::Requested,
+    );
+
+    // Run the "region replacement" task to transition the request to Running.
+
+    run_region_replacement(&internal_client).await;
+
+    let region_replacement = datastore
+        .get_region_replacement_request_by_id(&opctx, replacement_request_id)
+        .await
+        .unwrap();
+
+    assert_eq!(
+        region_replacement.replacement_state,
+        RegionReplacementState::Running,
```

Comment:
When this task is running, will it be doing the replacement in the background?

Reply:
The Running state is the state where the Drive saga can start from, but if the request is in state Running then nothing is currently happening.